config/samples/dataplane_v1beta1_openstackdataplanenodeset_baremetal_with_ipam.yaml

```diff
@@ -0,0 +1,154 @@
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+metadata:
+  name: openstack-edpm-ipam
+spec:
+  env:
+    - name: ANSIBLE_FORCE_COLOR
+      value: "True"
+    - name: ANSIBLE_ENABLE_TASK_DEBUGGER
+      value: "True"
+  services:
+    - configure-network
+    - validate-network
+    - install-os
+    - configure-os
+    - run-os
+    - ovn
+    - libvirt
+    - nova
+  deployStrategy:
+      deploy: true
+  baremetalSetTemplate:
+    deploymentSSHSecret: dataplane-ansible-ssh-private-key-secret
+    bmhLabelSelector:
+      app: openstack
+    ctlplaneInterface: enp1s0
+    dnsSearchDomains:
+      - osptest.openstack.org
+    cloudUserName: root
+  nodeTemplate:
+    ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+    nodes:
+       edpm-compute-0:
+         hostName: edpm-compute-0
+         ansible:
+           ansibleUser: root
+           ansibleHost: 192.168.122.100
+           ansibleVars:
+             ctlplane_ip: 192.168.122.100
+             internal_api_ip: 172.17.0.100
+             storage_ip: 172.18.0.100
+             tenant_ip: 172.19.0.100
+             fqdn_internal_api: edpm-compute-0.example.com
+         networks:
+         - name: CtlPlane
+           subnetName: subnet1
+           defaultRoute: true
+           fixedIP: 192.168.122.100
+         - name: InternalApi
+           subnetName: subnet1
+         - name: Storage
+           subnetName: subnet1
+         - name: Tenant
+           subnetName: subnet1
+       edpm-compute-1:
+         hostName: edpm-compute-1
+         ansible:
+           ansibleUser: root
+           ansibleHost: 192.168.122.101
+           ansibleVars:
+             ctlplane_ip: 192.168.122.101
+             internal_api_ip: 172.17.0.101
+             storage_ip: 172.18.0.101
+             tenant_ip: 172.19.0.101
+             fqdn_internal_api: edpm-compute-1.example.com
+         networks:
```

Comment:
You either provide them in nodeTemplate or nodes (i.e when using fixedIP, not at both the places). This would be ignored.